src/prefect/cli/root.py

```diff
@@ -143,3 +153,288 @@ def display(object: dict, nesting: int = 0):
+            parameters[k] = v
+
+    if params is not None:
+        parameters = json.loads(params)
+
+    base_deploy["parameters"] = parameters
+
+    ## RUN BUILD AND PUSH STEPS
+    step_outputs = {}
+    for step in project["build"] + project["push"]:
+        step_outputs.update(run_step(step))
+
+    # set other CLI flags
+    if name:
+        base_deploy["name"] = name
+    if version:
+        base_deploy["version"] = version
+    if tags:
+        base_deploy["tags"] = tags
+    if description:
+        base_deploy["description"] = description
+
+    # TODO: add schedule
+
+    if work_pool_name:
+        base_deploy["work_pool"]["name"] = work_pool_name
+    if work_queue_name:
+        base_deploy["work_pool"]["work_queue_name"] = work_queue_name
+
+    ## apply templating from build and push steps to the final deployment spec
+    base_deploy["work_pool"]["job_variables"] = apply_values(
+        base_deploy["work_pool"]["job_variables"], step_outputs
+    )
+
+    async with prefect.get_client() as client:
+        flow_id = await client.create_flow_from_name(base_deploy["flow_name"])
+
+        deployment_id = await client.create_deployment(
+            flow_id=flow_id,
+            name=base_deploy["name"],
+            work_queue_name=base_deploy["work_pool"]["work_queue_name"],
+            work_pool_name=base_deploy["work_pool"]["name"],
+            version=base_deploy["version"],
+            schedule=base_deploy["schedule"],
+            parameters=base_deploy["parameters"],
+            description=base_deploy["description"],
+            tags=base_deploy["tags"],
+            path=base_deploy["path"],
+            entrypoint=base_deploy["entrypoint"],
+            parameter_openapi_schema=base_deploy["parameter_openapi_schema"].dict(),
+            pull_steps=project["pull"],
+            infra_overrides=base_deploy["work_pool"]["job_variables"],
+        )
+
+        app.console.print(
+            (
+                f"Deployment '{base_deploy['flow_name']}/{base_deploy['name']}'"
+                f" successfully created with id '{deployment_id}'."
+            ),
+            style="green",
+        )
+
+        if PREFECT_UI_URL:
+            app.console.print(
+                "View Deployment in UI:"
+                f" {PREFECT_UI_URL.value()}/deployments/deployment/{deployment_id}"
+            )
+
+        if base_deploy["work_pool"]["name"] is not None:
+            app.console.print(
+                "\nTo execute flow runs from this deployment, start a worker that"
+                f" pulls work from the {base_deploy['work_pool']['name']!r} work pool"
+            )
+        elif base_deploy["work_pool"]["work_queue_name"] is not None:
+            app.console.print(
+                "\nTo execute flow runs from this deployment, start a worker that"
+                " pulls work from the"
+                f" {base_deploy['work_pool']['work_queue_name']!r} work queue"
+            )
```

Comment:
If the intention is to use this command in conjunction with a worker, we probably want to error if `prefect deploy` is used without a work pool. Workers require a work pool to start up and shouldn't be polling the `default-agent-pool`.